src/types/capabilities.rs

```diff
@@ -31,21 +33,28 @@ use std::hash::Hash;
 pub struct Capabilities(
     // Note that this field isn't *actually* 'static.
     // Rather, it is tied to the lifetime of the `ZeroCopy` that contains this `Name`.
-    pub(crate) HashSet<&'static str>,
+    pub(crate) HashSet<Capability<'static>>,
 );
 
 impl Capabilities {
     /// Check if the server has the given capability.
-    pub fn has<S: ?Sized>(&self, s: &S) -> bool
-    where
-        for<'a> &'a str: Borrow<S>,
-        S: Hash + Eq,
-    {
+    pub fn has<'a>(&self, s: &Capability<'a>) -> bool {
         self.0.contains(s)
     }
 
+    /// Check if the server has the given capability via str.
+    pub fn has_str(&self, s: &str) -> bool {
+        if s == IMAP4REV1_CAPABILITY {
+            self.has(&Capability::Imap4rev1)
+        } else if s.starts_with(AUTH_CAPABILITY_PREFIX) {
+            self.has(&Capability::Auth(&s[AUTH_CAPABILITY_PREFIX.len()..]))
+        } else {
+            self.has(&Capability::Atom(s))
+        }
+    }
+
     /// Iterate over all the server's capabilities
-    pub fn iter(&self) -> Iter<&str> {
+    pub fn iter(&self) -> Iter<Capability> {
```

Comment:
Hmm, this is also backwards incompatible.. I would pull the same trick as above here. Add `iter_str`, deprecate the old `iter` (but with the same signature) pointing people to `iter_str`, and then add the iterator over `Capability` under the name `all`? With the next major version bump we'll promote `contains` to `has` and `all` to `iter` (unless we prefer the new names?).